### ISSUE TITLE
[BUG] apps with the same name not showing

### ISSUE THREAD

**Issue by KhaiCo** [samolego/Canta#21]
**Describe the bug**
If there is an already existing app which have the same name with an existing app, only one app will appear. This issue is present in both the installed app menu (left hand side on canta) and the uninstalled (right hand side on canta).

https://github.com/samolego/Canta/assets/130490107/b5210197-c25c-41ae-9b1e-e4c412bd4e93

^as you can see in the above video I have 2 calculator app—but when I searched for it, only the new app shows up with the old app (system) not showing up. But after uninstalling new app, the old app can be observed again after an app restart. This issue also exist in the uninstalled menu, where if there is an uninstalled app with the same name as an installed app, after an app restart it'll disappear with no way of getting it back and only the installed one remain.

**To Reproduce**
1. Install an app which have the same name with an already existing app (usually system app)
2. Search for the old app (existing app) to uninstall
3. Find that only the new app is found when searched for (in the list too)

**Expected behavior**
All apps showing correctly

**Version info**
 - Android version: 12
 - Canta version: 1.2.0
 - Shizuku version: 13.5.2.r1028.2bbf671

**Logcat**

<details>
  <summary>View logcat</summary>
  ```
[05_12-20-40-53_486.log]1701783481.144 10489  1988  1988 I samo_lego.cant: Using CollectorTypeCC GC.
1701783481.147 10489  1988  1988 E samo_lego.cant: Not starting debugger since process cannot load the jdwp agent.
1701783481.150 10489  1988  1988 D ProcessState: Binder ioctl to enable oneway spam detection failed: Invalid argument
1701783481.180 10489  1988  1988 D CompatibilityChangeReporter: Compat change id reported: 171979766; UID 10489; state: ENABLED
1701783481.199 10489  1988  2051 D AppScoutStateMachine: 1988-ScoutStateMachinecreated
1701783481.202 10489  1988  1988 D nativeloader: Configuring clns-4 for other apk /data/app/~~DYX_goLtekyG-myqw0aaMw==/org.samo_lego.canta-XaKIP_5P0nxGjXmC9QQ61w==/base.apk. target_sdk_version=33, uses_libraries=, library_path=/data/app/~~DYX_goLtekyG-myqw0aaMw==/org.samo_lego.canta-XaKIP_5P0nxGjXmC9QQ61w==/lib/arm64:/data/app/~~DYX_goLtekyG-myqw0aaMw==/org.samo_lego.canta-XaKIP_5P0nxGjXmC9QQ61w==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/org.samo_lego.canta
1701783481.211 10489  1988  1988 V GraphicsEnvironment: ANGLE Developer option for 'org.samo_lego.canta' set to: 'default'
1701783481.211 10489  1988  1988 V GraphicsEnvironment: Updatable production driver is not supported on the device.
1701783481.212 10489  1988  1988 I ForceDarkHelperStubImpl: initialize for org.samo_lego.canta , ForceDarkOrigin
1701783481.213 10489  1988  1988 D OpenGLRenderer: JNI_OnLoad success
1701783481.213 10489  1988  1988 I MiuiForceDarkConfig: setConfig density:2.750000, mainRule:0, secondaryRule:0, tertiaryRule:0
1701783481.213 10489  1988  1988 D NetworkSecurityConfig: No Network Security Config specified, using platform default
1701783481.214 10489  1988  1988 D NetworkSecurityConfig: No Network Security Config specified, using platform default
1701783481.214 10489  1988  1988 D ShizukuProvider: Initialize Sui: false
1701783481.216 10489  1988  2051 D ShizukuProvider: binder received
1701783481.218 10489  1988  2042 D ShizukuProvider: sendBinder is called when already a living binder
1701783481.218 10489  1988  2051 I ShizukuApplication: attachApplication
1701783481.268 10489  1988  1988 W samo_lego.canta: type=1400 audit(0.0:86678): avc: denied { read } for name="max_map_count" dev="proc" ino=11396113 scontext=u:r:untrusted_app:s0:c233,c257,c512,c768 tcontext=u:object_r:proc_max_map_count:s0 tclass=file permissive=0 app=org.samo_lego.canta
1701783481.277 10489  1988  1988 I AdrenoGLES-0: QUALCOMM build                   : db3d445dbc, Ia06b22fa1a
1701783481.277 10489  1988  1988 I AdrenoGLES-0: Build Date                       : 10/04/21
1701783481.277 10489  1988  1988 I AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.32.02.16
1701783481.277 10489  1988  1988 I AdrenoGLES-0: Local Branch                     :
1701783481.277 10489  1988  1988 I AdrenoGLES-0: Remote Branch                    :
1701783481.277 10489  1988  1988 I AdrenoGLES-0: Remote Branch                    :
1701783481.277 10489  1988  1988 I AdrenoGLES-0: Reconstruct Branch               :
1701783481.277 10489  1988  1988 I AdrenoGLES-0: Build Config                     : S P 10.0.7 AArch64
1701783481.277 10489  1988  1988 I AdrenoGLES-0: Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
1701783481.280 10489  1988  1988 I AdrenoGLES-0: PFP: 0x016dd093, ME: 0x00000000
1701783481.295 10489  1988  1988 D ViewScrollStubImpl: the value of ScrollFrictionRes is: 0.015
1701783481.307 10489  1988  1988 D IS_CTS_MODE: false
1701783481.307 10489  1988  1988 D MULTI_WINDOW_SWITCH_ENABLED: false
1701783481.308 10489  1988  1988 D IS_CTS_MODE: false
1701783481.308 10489  1988  1988 D MULTI_WINDOW_SWITCH_ENABLED: false
1701783481.309 10489  1988  1988 D DecorView[]: getWindowModeFromSystem  windowmode is 1
1701783481.325 10489  1988  1988 D DecorView[]: getWindowModeFromSystem  windowmode is 1
1701783481.325 10489  1988  1988 D DecorView[]: updateDecorCaptionStatus displayWindowDecor is false
1701783481.351 10489  1988  1988 D SurfaceView: UPDATE null, mIsCastMode = false
1701783481.354 10489  1988  2062 D hw-ProcessState: Binder ioctl to enable oneway spam detection failed: Invalid argument
1701783481.362 10489  1988  1988 D SurfaceView: UPDATE Surface(name=SurfaceView[org.samo_lego.canta/org.samo_lego.canta.MainActivity])/@0xd61b2d5, mIsProjectionMode = false
1701783481.362 10489  1988  1988 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
1701783481.362 10489  1988  1988 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
1701783481.362 10489  1988  1988 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
1701783481.370 10489  1988  1988 D DecorView[]: getWindowModeFromSystem  windowmode is 1
1701783481.370 10489  1988  1988 D DecorView[]: updateDecorCaptionStatus displayWindowDecor is false
1701783481.397 10489  1988  2076 E LB      : fail to open file: No such file or directory
1701783481.438 10489  1988  1988 D DecorView[]: onWindowFocusChanged hasWindowFocus true
1701783482.085 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.miui.screenrecorder/2.png: open failed: ENOENT (No such file or directory)
1701783482.180 10489  1988  2207 D IconCustomizer: Content Ratio = -1.0
1701783482.180 10489  1988  2207 D IconCustomizer: Generate customized icon for com.google.android.ext.services.png
1701783482.623 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.android.updater/2.png: open failed: ENOENT (No such file or directory)
1701783482.870 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.android.providers.downloads/2.png: open failed: ENOENT (No such file or directory)
1701783483.173 10489  1988  2032 W samo_lego.cant: Cleared Reference was only reachable from finalizer (only reported once)
1701783483.279 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.android.soundrecorder/2.png: open failed: ENOENT (No such file or directory)
1701783483.394 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.mi.android.globallauncher/2.png: open failed: ENOENT (No such file or directory)
1701783483.438 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.android.providers.downloads.ui/2.png: open failed: ENOENT (No such file or directory)
1701783483.817 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.android.contacts/2.png: open failed: ENOENT (No such file or directory)
1701783484.372 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.android.calendar/2.png: open failed: ENOENT (No such file or directory)
1701783486.042 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.google.android.ims/2.png: open failed: ENOENT (No such file or directory)
1701783486.365 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.xiaomi.mipicks/2.png: open failed: ENOENT (No such file or directory)
1701783486.982 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.miui.calculator/2.png: open failed: ENOENT (No such file or directory)
1701783488.424 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.android.providers.contacts/2.png: open failed: ENOENT (No such file or directory)
1701783488.496 10489  1988  2036 W System  : A resource failed to call close.
1701783490.065 10489  1988  1988 W MirrorManager: this model don't Support
1701783490.407 10489  1988  1988 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.b{acbbe2e VFED..... .F....ID 0,0-1080,2270 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
1701783490.418 10489  1988  1988 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.b{acbbe2e VFED..... .F....ID 0,0-1080,2270 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
1701783490.432 10489  1988  1988 I AssistStructure: Flattened final assist data: 452 bytes, containing 1 windows, 3 views
1701783490.488 10489  1988  1988 D InsetsController: show(ime(), fromIme=true)
1701783490.502 10489  1988  1988 D InsetsController: show(ime(), fromIme=true)
1701783491.075 10489  1988  1988 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.b{acbbe2e VFED..... .F...... 0,0-1080,2270 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
1701783491.080 10489  1988  1988 D InsetsController: show(ime(), fromIme=true)
1701783491.198 10489  1988  1988 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.b{acbbe2e VFED..... .F...... 0,0-1080,2270 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
1701783491.201 10489  1988  1988 D InsetsController: show(ime(), fromIme=true)
1701783491.516 10489  1988  1988 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.b{acbbe2e VFED..... .F...... 0,0-1080,2270 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
1701783491.519 10489  1988  1988 D InsetsController: show(ime(), fromIme=true)
1701783492.740 10489  1988  1988 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.b{acbbe2e VFED..... .F...... 0,0-1080,2270 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
1701783492.748 10489  1988  1988 D InsetsController: show(ime(), fromIme=true)
1701783493.279 10489  1988  1988 D DecorView[]: onWindowFocusChanged hasWindowFocus false
1701783485.044 10489  1988  2207 E BitmapFactory: Unable to decode stream: java.io.FileNotFoundException: /system/media/theme/miui_mod_icons/dynamic/com.duokan.phone.remotecontroller/2.png: open failed: ENOENT (No such file or directory))
  ```
</details>

**Comment by samolego** [comment]
Ah I think I'm using a set instead of list ...

**Comment by samolego** [comment]
This will probably be resolved in next app version :). If you want & have the know-hpw, you can try building current master source.

**Comment by samolego** [comment]
Fixed in 2.0.0
